core/src/main/scala/org/broadinstitute/dsde/rawls/deltalayer/DeltaLayerTranslator.scala

```diff
@@ -0,0 +1,132 @@
+  val ERR_EMPTY_DATATYPE = "This method requires at least one string, boolean, or numeric scalar value to update."
+
+  /**
+   * Inspects the incoming EntityUpdateDefinitions and throws exceptions if any of those updates
+   * are illegal.
+   * @param entityUpdates the updates to validate
+   * @return the original set of updates, if all are valid
+   */
+  def validateEntityUpdates(entityUpdates: Seq[EntityUpdateDefinition]): Seq[EntityUpdateDefinition] = {
+    // verify that all entity names are legal UUIDs - they should be datarepo_row_ids
+    val uuidAttempts = entityUpdates.map(upd => Try(UUID.fromString(upd.name)))
+    if (uuidAttempts.exists(_.isFailure))
+      throw new DeltaLayerException(ERR_INVALID_ENTITYNAME, code = BadRequest)
+
+    // verify that all operations are AddUpdateAttribute only
+    val allOps: Seq[AttributeUpdateOperation] = entityUpdates flatMap { upd => upd.operations }
+
+    if (allOps.isEmpty)
+      throw new DeltaLayerException(ERR_EMPTY_OPERATIONS, code = BadRequest)
+
+    val (upserts, others) = allOps.partition {
+      case _:AddUpdateAttribute => true
+      case _ => false
+    }
+
+    if (others.nonEmpty)
+      // in the future, may want to provide more info such as entity type/name and illegal operation
+      throw new DeltaLayerException(ERR_INVALID_OPERATION, code = BadRequest)
+
+    if (upserts.isEmpty)
+      // how did we reach here? This should be impossible
+      throw new DeltaLayerException(ERR_EMPTY_OPERATIONS, code = BadRequest)
+
+    // validate that all upsert values are string/boolean/number only
+    val allValues = allOps.collect {
+      case a:AddUpdateAttribute => a.addUpdateAttribute
+    }
+
+    if (allValues.size != allOps.size)
+      throw new DeltaLayerException("Unexpected error: count of update values did not match count of update operations",
+        code = InternalServerError)
+
+    val (supportedTypes, unsupportedTypes) = allValues partition isSupportedDataType
+
+    if (unsupportedTypes.nonEmpty)
+      throw new DeltaLayerException(ERR_INVALID_DATATYPE, code = BadRequest)
+
+    if (supportedTypes.isEmpty)
+      // how did we reach here? This should be impossible
+      throw new DeltaLayerException(ERR_INVALID_DATATYPE, code = BadRequest)
+
+    // everything validated; return the original update definitions
+    entityUpdates
+  }
+
+  /**
+   * is the supplied Attribute supported by Delta Layer?
+   * @param attr the Attribute to inspect
+   * @return whether or not Delta Layer supports this Attribute's type
+   */
+  private def isSupportedDataType(attr: Attribute): Boolean = attr match {
+    case AttributeBoolean(_) => true
+    case AttributeNumber(_) => true
+    case AttributeString(_) => true
+    case AttributeNull => true
+    case AttributeValueEmptyList => true
+    case AttributeValueList(elems) => elems forall isSupportedDataType
+    case _ =>
```

Comment:
Could we combine cases like: `case AttributeBoolean(_) | AttributeNumber(_) | AttributeString(_) | AttributeNull | AttributeValueEmptyList => true`

Reply:
done! I listed them all out for readability … my idea of readability may be in the minority :)